packages/core/nuxt-theme-module/theme/components/LoginModal.vue

```diff
@@ -183,20 +191,49 @@ export default {
   setup() {
     const { isLoginModalOpen, toggleLoginModal } = useUiState();
     const form = ref({});
+    const serverError = ref({});
     const isLogin = ref(false);
     const createAccount = ref(false);
     const rememberMe = ref(false);
-    const { register, login, loading } = useUser();
+    const { register, login, loading, error } = useUser();
 
-    watch(isLoginModalOpen, () => {
+    watch([isLoginModalOpen, isLogin], (newValue, prevValues) => {
       if (isLoginModalOpen) {
         form.value = {};
+        serverError.value = {};
       }
+      if (newValue[1] !== prevValues[1]) serverError.value = {};
     });
 
+    const handleError = ({ email }) => {
```

Comment:
It's worth externalizing the `knownErrors` object so it can be easily overridden by theme integrators (`theme/helpers/errors/authentication.ts`)